pandas/core/indexes/category.py

```diff
@@ -324,20 +324,19 @@ def _reverse_indexer(self):
     @Appender(_index_shared_docs['__contains__'] % _index_doc_kwargs)
     def __contains__(self, key):
         hash(key)
-
-        if self.categories._defer_to_indexing:
-            return key in self.categories
-
-        return key in self.values
```

Comment:
Remind me: why do we need to check membership in `self.values` anymore?

Reply:
For indices, their indexing engine (i.e. ``._engine``) has a ``__contains__`` method which does the same thingbut  is faster.

Reply:
Awesome, thanks for clarifying!